### PR TITLE
Emoji Input 🎉

### PR DESCRIPTION
**Pull request by seancrater**
The emoji input feels good to go to me, personally would like to fork their repo and adjust some things that can't be changed via props.

With `deviceType` prop set to Android:
image

With `deviceType` prop set to iOS:
image

### CODE REVIEW

**Comment by igirard** [comment]
@seancrater What does this look like within the context of the actual creative editor screens? Can we add a little padding so it's not tight up against the top/right borders of the input box?

**Comment by seancrater** [comment]
@igirard This mockup probably isn't the best representation of it haha, in the in-app/push builders there will be padding on the inputs (the icon centers vertically and is offset right by 8px), can add a prop to adjust that if we'd like.

**Comment by dkraytsberg** [comment]
@seancrater I think building in good lookin' defaults is in our best interest, and if the implementing code wants to override that, power to them. So maybe add some padding and bump up the font size a little.

File: package.json [hunk 1]
```diff
@@ -89,5 +89,8 @@
     "style-loader": "0.13.1",
     "webpack": "2.2.1",
     "webpack-dev-server": "2.4.2"
+  },
+  "dependencies": {
+    "emoji-mart": "^1.0.0"
```
Comment (ntenczar) [inline package.json]: we're going to need to make this a peerDependency for dashboard

File: lib/Input.jsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+          name={ name }
+          type="text"
+          value={ this.state.inputValue }
+        />
+
+        { this.props.emojis ?
+          <button className="InputEmojiPicker__button" onClick={ this.toggleEmojiPicker }>
+            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 300 300">
+              <path
+                d="M150 229.9c-29.9 0-57.4-16.9-70.1-42.9-2.3-4.7-.3-10.3 4.3-12.6 4.7-2.3 10.3-.4 12.6
```
Comment (dkraytsberg) [inline lib/Input.jsx]: I think the multi-line strings in js are sometimes wonky, using a template literal may be safer

**Comment by seancrater** [comment]
image

Updated default padding/font-size to give a sense of desired space on these. Also default font size is 16 to prevent scaling if we ever decide to use this for a responsive use.

File: lib/Input.jsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    this.state = {
+      emojiPicker: false,
+      emojiPickerActive: false,
+      inputValue: ''
+    };
+  }
+
+  addEmoji(emoji) {
+    const inputValue = this.state.inputValue;
+    const position = this.state.inputPosition;
```
Comment (seancrater) [inline lib/Input.jsx]: Awesome catch, thank you!

**Comment by seancrater** [comment]
@dkraytsberg Fixed up the failing tests & issue that @ntenczar pointed out with duplication occurring

**Comment by ntenczar** [comment]
also you can leave the "page" as "Inputs" which is fine, and eventually we can have a few different input components on that page, not just emoji inputs (sorry for confusing)

**Comment by seancrater** [comment]
@ntenczar Updated, all good! 👍
Let me know if there's anything else we want to change before trying this out in dash.

File: lib/EmojiInput.jsx [hunk 4]
```diff
@@ -165,6 +165,7 @@ class EmojiInput extends React.Component {
 }
 
 EmojiInput.propTypes = {
+  /** Modifier class i.e. EmojiInput--right-aligned */
```
Comment (ntenczar) [inline lib/EmojiInput.jsx]: might be even better to just have an `alignment` prop that defaults to centered, but can take either left or right

Comment (seancrater) [inline lib/EmojiInput.jsx]: Now we're talking! I'll get that added now

**Comment by seancrater** [comment]
giphy

**Pull request merged**